### PR TITLE
Refactor reenqueues and waits

### PR DESCRIPTION
**Pull request by rmweir**
**Issue:**
https://github.com/rancher/rancher/issues/28054

### CODE REVIEW

**Review by brendarearden — DISMISSED** [review]
👍

**Comment by cbron** [comment]
@rmweir this one is going to get refactored right ? Removing reviewers until then.

File: pkg/controllers/management/eks/eks_cluster_handler.go [hunk 1]
```diff
@@ -385,44 +385,57 @@ func (e *eksOperatorController) updateEKSClusterConfig(cluster *mgmtv3.Cluster,
...
 
+// generateAndSetServiceAccount reads the EKSClusterConfig's secret once available and uses its fields to generate a
+// service account token. The token, CA cert, and API endpoint are then copied to the cluster status.
 func (e *eksOperatorController) generateAndSetServiceAccount(cluster *mgmtv3.Cluster) (*mgmtv3.Cluster, error) {
-	// service account token and API endpoint are need to deploy cluster agent, should be able to retrieve
-	// from secret
-	for i := 1; i < 15; i++ {
-		caSecret, err := e.secretsCache.Get(namespace.GlobalNamespace, cluster.Name)
+	backoff := wait.Backoff{
+		Duration: 2 * time.Minute,
```
Comment (dramich) [inline pkg/controllers/management/eks/eks_cluster_handler.go:392]:
```suggestion
		Duration: 2 * time.Seconds,
```

File: pkg/controllers/management/eks/eks_cluster_handler.go [hunk 2]
```diff
@@ -190,7 +190,7 @@ func (e *eksOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 			}
 		}
 
-		e.clusterEnqueueAfter(cluster.Name, 20*time.Second)
+		e.clusterEnqueueAfter(cluster.Name, 5*time.Second)
```
Comment (dramich) [inline pkg/controllers/management/eks/eks_cluster_handler.go:193]: Turning your enqueue time into a const would make updates easier

File: pkg/controllers/management/eks/eks_cluster_handler.go [hunk 3]
```diff
@@ -385,44 +385,57 @@ func (e *eksOperatorController) updateEKSClusterConfig(cluster *mgmtv3.Cluster,
...
-		endpoint := string(caSecret.Data["endpoint"])
-		ca := string(caSecret.Data["ca"])
-		saToken, err := generateSAToken(sess, cluster.Spec.EKSConfig.DisplayName, endpoint, ca)
-		if err != nil {
-			return cluster, err
-		}
+	logrus.Infof("generating service account token for cluster [%s]", cluster.Name)
+	sess, _, err := controller.StartAWSSessions(e.secretsCache, *cluster.Spec.EKSConfig)
+	if err != nil {
+		return cluster, nil
```
Comment (dramich) [inline pkg/controllers/management/eks/eks_cluster_handler.go:419]: Why is the error not being returned?

Comment (rmweir) [reply pkg/controllers/management/eks/eks_cluster_handler.go:419]: mistake

**Review by dramich — APPROVED** [review]

**Pull request merged**